client/txn.go

```diff
@@ -302,10 +310,11 @@ func endTxnCall(commit bool) proto.Call {
 func (txn *Txn) exec(retryable func(txn *Txn) error) (err error) {
 	// Run retryable in a retry loop until we encounter a success or
 	// error condition this loop isn't capable of handling.
+
 	for r := retry.Start(txn.db.txnRetryOptions); r.Next(); {
 		txn.haveTxnWrite, txn.haveEndTxn = false, false // always reset before [re]starting txn
 		if err = retryable(txn); err == nil {
-			if !txn.haveEndTxn && txn.haveTxnWrite {
+			if !txn.haveEndTxn && txn.txn.Writing {
```

Comment:
no. `Writing` is for tracking whether the coordinator has a goroutine running for this transaction. That is the case **after** the first successful write. The client needs to track whether it's **going** to write. Only putting both together gives you precise info.
Feel free to try to change any of the above, but it's a more involved change outside the scope of this PR and I think it won't do much good (I looked into it).